src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { FC, useEffect, useMemo, useState } from 'react';
+import { Person } from '../types/Person';
+import { formatString } from '../servises/formatString';
+import debounce from 'lodash.debounce';
+
+type Props = {
+  people: Person[];
+  onSelected: (person: Person | null) => void;
+  delay?: number;
+};
+
+export const Autocomplete: FC<Props> = ({
+  people,
+  onSelected,
+  delay = 300,
+}) => {
+  const [query, setQuery] = useState('');
+  const [isDropdownVisible, setIsDropdownVisible] = useState(false);
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useMemo(() => {
+    return debounce(setAppliedQuery, delay);
+  }, [delay]);
+
+  useEffect(() => {
+    return () => applyQuery.cancel();
+  }, [applyQuery]);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+  };
+
+  const filteredPeople = useMemo(() => {
+    if (appliedQuery.trim().length === 0) {
+      return people;
+    }
+
+    return people.filter(person =>
+      formatString(person.name).includes(appliedQuery.toLowerCase()),
```

Comment:
The `formatString` function currently does not modify the string as intended. To normalize spaces, consider using a regular expression to replace multiple spaces with a single space, like `str.replace(/\s+/g, ' ')`. This will ensure that the filtering logic works correctly with normalized input strings.